apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -436,28 +437,22 @@ handle_request('GetContractPoI', Req, _Context) ->
     process_request(ParseFuns, Req);
 
 handle_request('GetAccountBalance', Req, _Context) ->
-    case aec_base58c:safe_decode(account_pubkey, maps:get('account_pubkey', Req)) of
-        {ok, AccountPubkey} ->
-            case get_block_hash_optionally_by_hash_or_height(Req) of
-                {error, not_found} ->
-                    {404, [], #{reason => <<"Block not found">>}};
-                {error, invalid_hash} ->
-                    {400, [], #{reason => <<"Invalid block hash">>}};
-                {error, blocks_mismatch} ->
-                    {400, [], #{reason => <<"Invalid height and hash combination">>}};
-                {ok, Hash} ->
-                      case aehttp_logic:get_account_balance_at_hash(AccountPubkey, Hash) of
+    ParseFuns = [read_required_params([address]),
+                 base58_decode([{address, address, [account_pubkey,
+                                                    contract_pubkey]}]),
+                 get_block_hash_optionally_by_hash_or_height(block_hash),
+                 fun(_, #{address := Pubkey, block_hash := AtHash}) ->
+                      case aehttp_logic:get_account_balance_at_hash(Pubkey, AtHash) of
                           {error, account_not_found} ->
-                              {404, [], #{reason => <<"Account not found">>}};
+                              {error, {404, [], #{reason => <<"Account not found">>}}};
```

Comment:
must be some magic going on, but in your test you match on {ok, {404, ...}}... but that's the request function manipulating it, I guess.

Reply:
`aehttp_helpers:process_request/2` runs a list of functions on top of the incoming Req and aggregates state in a map of its own.
If any function ever returns an `{error, CowboyResponse}` - the chain is broken (the rest functions are not being applied at all) and the `CowboyResponse` is sent to the client.

The `ok` in the test has nothing to do with this whatsoever - it just states that we've managed to make a request to the URL...